### PR TITLE
rgw: fix input args for function UpdateIndex::complete_del

### PR DESCRIPTION
**Pull request by Huber-ming**
fix: https://tracker.ceph.com/issues/55619

Signed-off-by: Huber-ming <zhangsm01@inspur.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Huber-ming** [comment]
jenkins test windows

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -5270,19 +5270,23 @@ int RGWRados::Object::Delete::delete_obj(optional_yield y, const DoutPrefixProvi
...
+  index_op.get_bucket_shard(&bs, dpp);
+  auto& index_ref = bs->bucket_obj.get_ref();
+
   r = rgw_rados_operate(dpp, ioctx, ref.obj.oid, &op, null_yield);
 
   /* raced with another operation, object state is indeterminate */
   const bool need_invalidate = (r == -ECANCELED);
 
-  int64_t poolid = ioctx.get_id();
+  int64_t poolid = index_ref.pool.ioctx().get_id();
```
Comment (cbodley) [inline src/rgw/rgw_rados.cc:5282]: @Huber-ming can you please help me understand why this change is needed?

it looks like cls_rgw is expecting to get these values from the data pool where the head object resides, not from the index pool. `RGWRados::Object::Write::_do_write_meta()` gets these from the head object's pool as well: https://github.com/ceph/ceph/blob/master/src/rgw/rgw_rados.cc#L3239-L3240

**Review by cbodley — CHANGES_REQUESTED** [review]
